src/prss.rs

```diff
@@ -7,18 +7,35 @@ use byteorder::{ByteOrder, LittleEndian};
 use hkdf::Hkdf;
 use rand::{CryptoRng, RngCore};
 use sha2::Sha256;
+use std::ops::Index;
+use std::{marker::PhantomData, mem::size_of};
 use x25519_dalek::{EphemeralSecret, PublicKey};
 
-/// A participant in a 2-of-3 replicated secret sharing.
+pub trait SpaceIndex {
+    const MAX: usize;
```

Comment:
what do you think about using a fixed-size integer type for it or a static assert that enforces a minimum size for `usize` for this program to work correctly?

Reply:
It's harder to use the value as an index without that.  I think that it's OK to just deal with the weirdness here, which the compiler should be able to optimize out completely.

Reply:
Yep it looks like we don't bake any assumptions here about `usize` being 8 byte long, so it is probably fine not to introduce static compilation checks here